pilot/pkg/model/context.go

```diff
@@ -641,11 +648,6 @@ func ParseServiceNodeWithMetadata(s string, metadata *NodeMetadata) (*Proxy, err
 
 // ParseIstioVersion parses a version string and returns IstioVersion struct
 func ParseIstioVersion(ver string) *IstioVersion {
-	if strings.HasPrefix(ver, "master-") {
-		// This proxy is from a master branch build. Assume latest version
-		return MaxIstioVersion
-	}
-
 	// strip the release- prefix if any and extract the version string
 	ver = istioVersionRegexp.FindString(strings.TrimPrefix(ver, "release-"))
```

Comment:
@howardjohn does Istio Version for release start with "release-" ?

Reply:
we used to, same with master-, until 1.4

Reply:
ok. I think we should leave it like that for now and may be remove later. 1.5 pilot still should process "release-" envoys.